sonic_platform_base/sonic_sfp/sfputilhelper.py

```diff
@@ -45,6 +47,25 @@ class SfpUtilHelper(object):
     def __init__(self):
         pass
 
+    def get_port_mapping(self, asic_inst):
+        try:
+            if multi_asic.is_multi_asic():
+                ports_data = multi_asic.get_port_table_for_asic("{}{}".format(multi_asic.ASIC_NAME_PREFIX, asic_inst))
+            else:
+                ports_data = multi_asic.get_port_table_for_asic(multi_asic.DEFAULT_NAMESPACE)
+        except Exception as e:
+            return None
+
+        ports = ast.literal_eval(json.dumps(ports_data))
+        for port in ports.keys():
+            if "index" not in ports[port]:
+                print("Failed to get port config: invalid port entry {}: index field doesn't exist".format(port),
+                    file=sys.stderr
+                )
+                return None
+
+        return ports
+
     def read_porttab_mappings(self, porttabfile, asic_inst=0):
```

Comment:
There are lot of duplicate code in this function. This fn is essentially creating various port related mapping. [get_port_config](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/portconfig.py#L99) tries to read the config_db (provide the right asic name for multi-asic) and if that fails then it reads platform.json and finally reads port_config.ini. can you make use of get_port_config()'s returned port{} map to generate the mapping as that will remove lot of duplicate code from read_porttab_mappings()

Reply:
@prgeor this is a very good initiative, but i would suggest to have a separate PR for that purpose, since it sounds for me like a refactoring activity. We would like to have this fix as is, since it is very important for us.

Now back to your proposal. From the code i can see, a couple of things needs to be done, IMHO:
1. No need to have everything in `get_port_config` - it is hard to read such a code
2. Separate each processing part with a dedicated method: `config_db`, `platform.json`, `port_config.ini`
3. Make sure `port_config.ini` part can handle both old/new style of the file

Please let me know what do you think.

Reply:
I believe the refactoring code should be done on master and not on this branch. Are the comments related to this branch only or to the code available on master? 
if the code in master is already fixed, I agree that the fix should be improved. If not, let's have it as is and focus on improvements on master as long as the change is working.